### PR TITLE
Avoid more local links and bad http requests

### PR DESCRIPTION
**Pull request by annando**
The main feature of this PR is a complete overwhaul of the "Announce" processing. Initially it had to be done in a quite weird way, since we hadn't supported real reshares at that point in time. But now we support it, so we can simplify the feature.

Also this PR contains some more improvements for handling with bad URL.

### CODE REVIEW

File: src/Model/APContact.php [hunk 1]
```diff
@@ -368,10 +368,13 @@ public static function getByURL(string $url, $update = null): array
 
 		// Unhandled from Kroeg
 		// kroeg:blocks, updated
```
Comment (MrPetovan) [inline src/Model/APContact.php]: What's this comment for?

Comment (annando) [reply src/Model/APContact.php]: This is some old stuff that I added when I did my first steps with AP.

Comment (MrPetovan) [reply src/Model/APContact.php]: Can you please remove it then?
```suggestion
```

Comment (annando) [reply src/Model/APContact.php]: Done

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**